### ISSUE TITLE
Alllow string replacement in bulk renaming

### ISSUE THREAD

**Issue by tsujan** [lxqt/pcmanfm-qt#1888]
For example, the user may want to replace "this" with "that" in files `this_file_1`, `this_file_2`, `this_file_3`,... without changing anything else.

The logic is simple — regex could be allowed too — but how to keep the GUI intuitive and uncluttered is another question. I'll think about it after 2.0.0 is released. Suggestions about the GUI (bulk renaming dialog) are welcome.

Self-assigning...

**Comment by stefonarch** [comment]
IMO - added do the numbering we have already:
* Replace `foo` with `gaa`
* case sensitive yes|no
* allow regex 
* transform lower|uppercase ?

**Comment by tsujan** [comment]
I agree with most of them, and they're relatively easy to implement. Only "previewing of renamed files" seems like an overkill to me. Switching between lowercase and uppercase doesn't seem very usable either. All in all, there's no problem on the coding side.

My "problem" is just finding the best way of including it inside the bulk renaming dialog, such that the resulting GUI becomes uncluttered and intuitive as far as possible.

**Comment by stefonarch** [comment]
It should be under "Tools" at the end IMO.

**Comment by tsujan** [comment]
A separate menu-item? IMHO, there's no need to that. Moreover, since it's about bulk renaming, the user expects to see it inside the "Bulk Rename" dialog.

Perhaps a separate tab in that dialog,....

**Comment by stefonarch** [comment]
It looks like a tool to me, like the .`desktop` editor.

**Comment by tsujan** [comment]
"Bulk Rename" is already under the "Edit" menu. None of the items under "Tools" edits existing things.

However, I have no problem with changing the place of "Bulk Rename" if really needed. My point is only that it's about bulk renaming, and so, it needs no separate menu-item.
